File: 258.Add Digits.cc
// https://leetcode.com/problems/add-digits/

class Solution {
public:
    int addDigits(int number) {
        std::ios::sync_with_stdio(false);
        cin.tie(nullptr);
        cout.tie(nullptr);
        
        int start = number;
        int acc = 0;
        while (start / 10 != 0)
        {
            acc = start;
            start = 0;
            while(acc != 0)
            {
                start += acc % 10;
                acc /= 10;
            }
        }
        
        return start;
    }
};

File: leetcode/0-300/251-300/258.Add Digits.rs
// https://leetcode.com/problems/add-digits§

impl Solution {
    pub fn add_digits(num: i32) -> i32 {
        let mut n = num;
        while n > 9 {
            let mut ncopy = n;
            let mut sum = 0;
            while ncopy > 0 {
                sum += ncopy % 10;
                ncopy /= 10;
            }
            n = sum;
        }
        n
    }
}